tests/failpoints/cases/test_split_region.rs

```diff
@@ -145,3 +145,161 @@ impl Filter for PrevoteRangeFilter {
+    configure_for_merge(&mut cluster);
+    cluster.cfg.raft_store.right_derive_when_split = true;
+    cluster.cfg.raft_store.apply_batch_system.max_batch_size = 1;
+    cluster.cfg.raft_store.apply_batch_system.pool_size = 2;
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+
+    let r1 = cluster.run_conf_change();
+    pd_client.must_add_peer(r1, new_peer(2, 2));
+    pd_client.must_add_peer(r1, new_peer(3, 3));
+
+    let mut region_a = pd_client.get_region(b"k1").unwrap();
+    // [-∞, k2), [k2, +∞)
+    //    b         a
+    cluster.must_split(&region_a, b"k2");
+
+    cluster.put(b"k0", b"v0").unwrap();
+    must_get_equal(&cluster.get_engine(3), b"k0", b"v0");
+
+    let region_b = pd_client.get_region(b"k0").unwrap();
+    let peer_b_1 = find_peer(&region_b, 1).cloned().unwrap();
+    cluster.must_transfer_leader(region_b.get_id(), peer_b_1);
+
+    let peer_b_3 = find_peer(&region_b, 3).cloned().unwrap();
+    assert_eq!(peer_b_3.get_id(), 1003);
+    let on_handle_apply_1003_fp = "on_handle_apply_1003";
+    fail::cfg(on_handle_apply_1003_fp, "pause").unwrap();
+    // [-∞, k1), [k1, k2), [k2, +∞)
+    //    c         b          a
+    cluster.must_split(&region_b, b"k1");
+
+    pd_client.must_remove_peer(region_b.get_id(), peer_b_3);
+    pd_client.must_add_peer(region_b.get_id(), new_peer(4, 4));
+
+    let mut region_c = pd_client.get_region(b"k0").unwrap();
+    let peer_c_3 = find_peer(&region_c, 3).cloned().unwrap();
+    pd_client.must_remove_peer(region_c.get_id(), peer_c_3);
+    pd_client.must_add_peer(region_c.get_id(), new_peer(4, 5));
+    // [-∞, k2), [k2, +∞)
+    //     c        a
+    pd_client.must_merge(region_b.get_id(), region_c.get_id());
+
+    region_a = pd_client.get_region(b"k2").unwrap();
+    let peer_a_3 = find_peer(&region_a, 3).cloned().unwrap();
+    pd_client.must_remove_peer(region_a.get_id(), peer_a_3);
+    pd_client.must_add_peer(region_a.get_id(), new_peer(4, 6));
+    // [-∞, +∞)
+    //    c
+    pd_client.must_merge(region_a.get_id(), region_c.get_id());
+
+    region_c = pd_client.get_region(b"k1").unwrap();
+    // [-∞, k2), [k2, +∞)
+    //     d        c
+    cluster.must_split(&region_c, b"k2");
+
+    let peer_c_4 = find_peer(&region_c, 4).cloned().unwrap();
+    pd_client.must_remove_peer(region_c.get_id(), peer_c_4);
+    pd_client.must_add_peer(region_c.get_id(), new_peer(3, 7));
+
+    cluster.put(b"k2", b"v2").unwrap();
+    must_get_equal(&cluster.get_engine(3), b"k2", b"v2");
+
+    fail::remove(on_handle_apply_1003_fp);
+
+    // If peer_c_3 is created, `must_get_none` will fail.
+    must_get_none(&cluster.get_engine(3), b"k0");
+}
+
+// Test if a peer is created from splitting when another initialized peer with the same
+// region id existed before and has been destroyed now.
+#[test]
+fn test_split_not_to_split_exist_tombstone_region() {
+    let mut cluster = new_node_cluster(0, 4);
```

Comment:
This is too complicated. How about:
1. transfer leader of region 1 to store 1
2. isolate store 3 from region 1
3. pause apply of replica on store 2 of region 1
4. split region 1 as region 1 and region 1000
5. destroy replica on store 3 of region 1000 after peer is initialized.
6. transfer leader of region 1 to replica on store 2 and remove transport filters.